Weekly view front end

SeanDonnelly1 opened the pull request:
-Added some CSS to the accordion to improve the overall look
-Merged Sagar's code to make tasks look more consistent with how they appear on other pages

SeanDonnelly1 committed:
Moved stuff around

Moved the dropdowns to be above the buttons on the page instead of below

SeanDonnelly1 committed:
Fixed login bug

SeanDonnelly1 committed:
Adjusted spacing for dropdowns

SeanDonnelly1 committed:
Removed comments

SeanDonnelly1 committed:
More minor adjustments

SeanDonnelly1 committed:
Attempted to add rounded borders

Added rounded borders to the dropdowns, but for some reason the dropdowns are now displayed inside each other. Will work on fixing this later.

SeanDonnelly1 committed:
Fixed border issue

Dropdowns are now displayed correctly and UI is working properly

SeanDonnelly1 committed:
Merge branch 'week-view-frontend'

ebradd5 committed:
added cursor pointer

SeanDonnelly1 committed:
Font changes

ebradd5 committed:
changes

ebradd5 committed:
changes

SeanDonnelly1 committed:
Update weekly.html

SeanDonnelly1 committed:
Added accordion

SeanDonnelly1 committed:
Update weekly.html

SeanDonnelly1 committed:
Moved CSS to styles.css and replaced dropdowns with accordion

sagarsubedi commented on templates/weekly.html:
```diff
@@ -3,62 +3,113 @@
 </div>
 
+
 <script>
```
Do we still need this?

sagarsubedi commented on templates/weekly.html:
```diff
@@ -3,62 +3,113 @@
-    <div class="d-flex float-start">
-        {% include 'add_task_button.html' %}
+
+<div class="container">
```
This accordion code is repeated a lot. Is there any way to minimize the repeated code?

SeanDonnelly1 replied:
I would have liked to use a loop or something for this, but each day in the accordion needs to have its own set of variables to work correctly. Otherwise, you'll have problems such as all sections of the accordion opening and closing at the same time when any one of them is clicked. I know the way I did it is messy, but I couldn't think of another way around this issue. Hopefully next sprint we can find some way to make this cleaner.

Qtipshooter replied:
Yea, looking at this for sure this can be made with a variable to assign things such as IDs or such.  Even if we just move some parts of it.  Why were multiple opening and closing at the same time?

Qtipshooter replied:
This was resolved with the loop if I recall

sagarsubedi commented on static/styles.css:
```diff
@@ -182,3 +182,46 @@ body{
 }
+
+/* Dropdown styles */
+  .dropdown {
```
Do we need any of these dropdown styles? I do not see any dropdown class in weekly.html file

sagarsubedi commented on static/styles.css:
```diff
@@ -182,3 +182,46 @@ body{
+    display:block;
+  }
+
+  .days {
```
There already seems to be a day class on calendar.html, could you clarify why this style exists?

SeanDonnelly1 replied:
I believe this was also used for the dropdowns. It was to apply CSS to the buttons for each day without changing how the content inside the dropdown would look. We most likely won't need this anymore, so I can delete this part

SeanDonnelly1 committed:
Deleted code that is no longer needed

SeanDonnelly1 committed:
Created a loop to set up the accordion

Qtipshooter commented on templates/weekly.html:
```diff
@@ -3,69 +3,43 @@
+      </h2>
+      <div id="collapse{{day}}" class="accordion-collapse collapse" aria-labelledby="heading{{day}}" data-bs-parent="#accordionWeek">
+        <div class="accordion-body">
+          <p>[Insert task here]</p>
```
The new accordion looks good while not repeating as much code.  Don't forget to add dummy data here as well

sagarsubedi commented on templates/weekly.html:
```diff
@@ -3,69 +3,43 @@
-        <button type="button" class="btn btn-yellow ms-3" onclick="location.href='/tasks';"><i
-                class="bi bi-calendar3-event"></i> Task View</button>
-    </div>
+  <h1>January 30 - February 5</h1>
```
@SeanDonnelly1 @ebradd5 Could we make this range of date dynamic depending on what week this is instead of just saying January 30 - February 5?

ebradd5 committed:
added current week heading to week-view

SeanDonnelly1 committed:
.

SeanDonnelly1 committed:
Played around with CSS

SeanDonnelly1 committed:
Merge branch 'main' into week-view-frontend

SeanDonnelly1 committed:
Update styles.css

SeanDonnelly1 committed:
Delete calendar-style.css

SeanDonnelly1 committed:
Merge branch 'main' into week-view-frontend

Qtipshooter commented on static/week-view.js:
```diff
@@ -0,0 +1,29 @@
+];
+
+//will return the first day of the week (Sunday)
+const firstDay = new Date(date.getTime() - 60*60*24*day*1000);
```
I'm kinda confused how this is returning the first day and last day of the week.  Could you explain the magic numbers?

Qtipshooter replied:
It still kinda has me confused.  From looking at it, I would assume that one 60 is minutes, one 60 is seconds, 24 is hours, but that doesn't make sense how it gets the Sunday of the week by multiplying that with the arbitrary number 1000 and the current number of the day.

SeanDonnelly1 commented on static/styles/styles.css:
```diff
@@ -225,4 +225,22 @@ body{
       background-color: var(--primary-yellow);
+    }
+    
+    .accordion {
```
I don't think we'll have an accordion on any other pages, so it should be fine to move it into its own file. I'll do that next time I work on it.

SeanDonnelly1 committed:
CSS Edits

-Created weekly.css for styles specific to weekly view page
-Added more CSS for accordion

SeanDonnelly1 committed:
Added formatting for tasks

SeanDonnelly1 committed:
Added "important" icon

Qtipshooter commented on templates/weekly.html:
```diff
@@ -1,71 +1,51 @@
+        <div id="collapse{{day}}" class="accordion-collapse collapse" aria-labelledby="heading{{day}}" data-bs-parent="#accordionWeek">
+          <div class="accordion-body">
+            <strong><i class="bi bi-brightness-high-fill" style="color:red"></i></strong>
+            <p class="task-header">yyyy-mm-dd | hh:mm</p>
```
Would it be a better idea to put these into some kind of div?  It looks like if there was more than one task that it would look funky and just kinda clash.  None of them have a second task on them though, so I am not sure.

sagarsubedi replied:
Yes, a div is exactly what I was suggesting - something to clearly differentiate multiple tasks in a single week. Also, a task taking up the entire width seemed a bit off.

Qtipshooter commented on static/scripts/week-view.js:
```diff
@@ -0,0 +1,29 @@
+];
+
+//will return the first day of the week (Sunday)
+const firstDay = new Date(date.getTime() - 60*60*24*day*1000);
```
I'm not sure what this code here is doing, there are just a bunch of magic numbers.  Would you be able to elaborate on what the numbers mean?  I think I can guess for all of them except the 1000, but it just seems so arbitrary and doesn't make sense.

sagarsubedi replied:
I agree. This is forcing a developer to look up what getTime() returns and then try to make sense of substracting the result of 60\*60\*24\*day\*1000 from getTime(). I know what getTime() returns but it is still unclear how this returns the firstDay of the week. Instead of explaining to us or putting a lengthy comment, it might be a good idea to clean up line 23 and 26. Just a suggestion.

sagarsubedi commented on static/styles/weekly.css:
```diff
@@ -0,0 +1,28 @@
+.accordion {
```
The new styles I have recommended might clash with some of the ones present here, just an FYI.

sagarsubedi reviewed (commented):
The only thing that would be functionally important here is how a task or group of tasks will look inside the accordion. Could you please work on that? And rest of the suggestions, you can ignore if you want or work on them as well. Rest aren't functional requirement

SeanDonnelly1 committed:
Added a box around each task entry

This will make it more readable when there are multiple tasks on the same day.

SeanDonnelly1 committed:
Adjusted the look of the accordion

ebradd5 committed:
js clarification

Qtipshooter commented on static/scripts/week-view.js:
```diff
@@ -0,0 +1,29 @@
+//will return the first day of the week in milliseconds (Sunday)
+const firstDay = new Date(date.getTime() - 60*60*24*currentDay*1000);
+
+//will return last day of the week in milliseconds (Saturday)
```
Oh, okay.  I think it makes a bit more sense now.  We can edit it to remove magic numbers later.

Qtipshooter reviewed (approved):
Okay, looks like it is something we can work with.  We just will have to make a new function that is slightly different than the old one.

sagarsubedi committed:
better looking tasks

SeanDonnelly1 committed:
Merge pull request #142 from Qtipshooter/week-view-frontend-new-task-view

different looking tasks under accordion on week view

sagarsubedi reviewed (approved):
Looks good. I think Sean still wanted to change the accordion background when clicked. Other than that, I think the front-end is now fine and the back-end person can start working on this.
